### PR TITLE
Add command truth table of ddr5

### PR DESCRIPTION
**Pull request by RRozak**

### CODE REVIEW

File: litedram/phy/ddr5/commands.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    # construct the value. CS is assumed to be H for 1st edge and L for 2nd edge.
+    # TODO: Fix currently fake truth table to be consistent with the JEDEC specs for DDR5
+    # Here, in contrary to LPDDR4's table, first element of array is related to CS low,
+    # the second to CS high. Something will have to be changed, because other parts of code
+    # rely on order from LPDDR4, but there are commands that have only row related to low.
+    # At this moment I keep it as it is now.
+    TRUTH_TABLE = {
+        "ACTIVATE":      ["L L L R0 R1 R2 R3 BA0 BA1 BG0 BG1 BG2 CID0 CID1 CID2",
+                          "R4 R5 R6 R7 R8 R9 R10 R11 R12 R13 R14 R15 R16, CID3/R17"],
+        "READ":          ["H L H H H BL*=L BA0 BA1 BG0 BG1 BG2 CID0 CID1 CID2",
```
Comment (acomodi) [inline litedram/phy/ddr5/commands.py]: I'd keep just `BL` instead of `BL*=L`. It is too verbose otherwise.

File: litedram/phy/ddr5/commands.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                          "C2 C3 C4 C5 C6 C7 C8 C9 C10 V H V V CID3"],
+        "WRITE":         ["H L H H L BL*=L BA0 BA1 BG0 BG1 BG2 CID0 CID1 CID2",
+                          "V C3 C4 C5 C6 C7 C8 C9 C10 V H WR_partial=L V CID3"],
```
Comment (acomodi) [inline litedram/phy/ddr5/commands.py]: I'd rename `WR_partial=L` to `WRP`. And add a comment below on what it means

File: litedram/phy/ddr5/commands.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    def parse_bit(self, bit, is_mrw):
+        assert len(self.dfi.bank) >= 8, "At least 8 DFI bankbits needed for Mode Register address"
+        assert len(self.dfi.address) >= 17, "At least 17 DFI addressbits needed for row address"
+        mr_address = self.dfi.bank if is_mrw else self.dfi.address
+        rules = {
+            "H":       lambda: 1,  # high
+            "L":       lambda: 0,  # low
+            "V":       lambda: 0,  # defined logic
+            "X":       lambda: 0,  # don't care
+            "CID3/R17": None,
```
Comment (acomodi) [inline litedram/phy/ddr5/commands.py]: I believe there cannot be `None` values.

Comment (RRozak) [reply litedram/phy/ddr5/commands.py]: There are only temporary. I will replace them with lambdas, when I will know what they should return

File: litedram/phy/ddr5/commands.py (hunk 1 above)
Comment (RRozak) [reply litedram/phy/ddr5/commands.py]: Done

File: litedram/phy/ddr5/commands.py (hunk 2 above)
Comment (RRozak) [reply litedram/phy/ddr5/commands.py]: Done

File: litedram/phy/ddr5/commands.py [hunk 4]
```diff
@@ -175,7 +175,8 @@ class Command(Module):
         "MRR":           ["H L H L H MRA0 MRA1 MRA2 MRA3 MRA4 MRA5 MRA6 MRA7 V",
                           "L L V V V V V V V V CW V V V"],
         "MRW":           ["H L H L L MRA0 MRA1 MRA2 MRA3 MRA4 MRA5 MRA6 MRA7 V",
-                          "OP0 OP1 OP2 OP3 OP4 OP5 OP6 OP7 V V CW V V V"]
+                          "OP0 OP1 OP2 OP3 OP4 OP5 OP6 OP7 V V CW V V V"],
+        "DESELECT":      ["X X X X X X X X X X X X X X"]
```
Comment (acomodi) [inline litedram/phy/ddr5/commands.py]: Given that deselect has all don't cares, we can handle it as previously, nothing should change there I think.

One thing to change though is the CS value, which is inverted, it is LOW in the first half of the transaction, and HIGH in the second, while for the `DESELECT` it is only HIGH

**Pull request closed without merging**